incubator/hnc/internal/validators/namespace.go

```diff
@@ -99,6 +106,16 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 	return allow("")
 }
 
+func (v *Namespace) illegalExcludedNamespaceLabel(req *nsRequest) admission.Response {
+	for l := range req.ns.Labels {
+		if l == api.LabelExcludedNamespace && !config.ExcludedNamespaces[req.ns.Name] {
+			msg := fmt.Sprintf("The namespace contains an illegal label %q, which should only be applied on the excluded namespaces set in the deployment configuration", api.LabelExcludedNamespace)
```

Comment:
This is a very low-level explanation of what's gone wrong. Maybe add some documentation and then reference that in the error message? I.e. don't call it an "illegal label," talk about the problem domain (e.g. "You can't exclude this namespace using the %q label from HNC because ..." or something like that).

Reply:
documented and linked.